src/app/app-routing.module.ts

```diff
@@ -1,7 +1,45 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
-
-const routes: Routes = [];
+import { LoginComponent } from './components/login/login.component';
+import { ProductsComponent } from './components/products/products.component';
+import { ShoppingCartComponent } from './components/shopping-cart/shopping-cart.component';
+import { authGuard } from './auth.guard';
+import { ProductDetailsComponent } from './components/product-details/product-details.component';
+import { CheckoutComponent } from './components/checkout/checkout.component';
+const routes: Routes = [
+  {
+    path: 'login',
+    component: LoginComponent
+  },
+  {
+    path: 'products',
+    component: ProductsComponent
+  },
+  {
+    path: 'product/:id',
+    component: ProductDetailsComponent,
+  },
+  {
+    path: 'product/:id/checkout',
+    component: CheckoutComponent,
+    canActivate: [authGuard]
+  },
+  {
+    path: 'checkout/:finalAmount',
+    component: CheckoutComponent,
+    canActivate: [authGuard]
+  },
+  {
```

Comment:
make cart public so user can add products to cart and to make purchase user has to login